src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/MarshalTests.cs

```diff
@@ -152,5 +152,134 @@ public static void GetHRForException()
+        {
+             if(PlatformDetection.IsWindows)
+             {
+                Assert.Throws<ArgumentNullException>(() => Marshal.GetHINSTANCE(null));
+             }
+             IntPtr ptr = Marshal.GetHINSTANCE(typeof(System.Int32).Module);
+             IntPtr ptr2 = Marshal.GetHINSTANCE(typeof(System.String).Module);
+             Assert.Equal(ptr, ptr2);;
+        }
+
+        [Fact]
+        public static void GetIDispatchForObject()
+        {
+            Assert.Throws<PlatformNotSupportedException>(() => Marshal.GetIDispatchForObject(null));   
+        }
+                
+        [Fact]
+        public static void GetTypedObjectForIUnknown()
+        {
+            if(PlatformDetection.IsWindows)
+            {
+                Assert.Throws<ArgumentNullException>(() => Marshal.GetTypedObjectForIUnknown((System.IntPtr)0, typeof(System.Int32)));
+            }  
+            else 
+            {
+                Assert.Throws<PlatformNotSupportedException>(() => Marshal.GetTypedObjectForIUnknown((System.IntPtr)0, typeof(System.Int32)));
+            }
+        }
+
+        [Fact]
+        public static void Prelink()
+        {
+            Assert.Throws<ArgumentNullException>(() => Marshal.Prelink(null));
+        }
+
+        [Fact]
+        public static void PrelinkAll()
+        {
+            Assert.Throws<ArgumentNullException>(() => Marshal.PrelinkAll(null));
+        }
+        
+        [Fact]
+        public static void PtrToStringAuto()
+        {
+            Assert.Equal(null, Marshal.PtrToStringAuto((IntPtr)0));
+        }
+
+        [Fact]
+        public static void SetComObjectData()
+        {
+             Assert.Throws<PlatformNotSupportedException>(() => Marshal.SetComObjectData(null, null, null));        
+        }
+
+        [Fact]
+        public static void StringToCoTaskMemAuto()
+        {
+            String s = null;
+
+            // passing null string should return 0
+            Assert.Equal((long)Marshal.StringToCoTaskMemAuto(s), 0);        
+            
+            s = "Hello World";
+            IntPtr ptr = Marshal.StringToCoTaskMemAuto(s);
+            String s2 = Marshal.PtrToStringAuto(ptr);
+            Assert.Equal(s, s2);  
+        }
+
+        [Fact]
+        public static void StringToHGlobalAuto()
+        {
+            String s = null;
+
+            // passing null string should return 0
+            Assert.Equal((long)Marshal.StringToHGlobalAuto(s), 0);        
+            
+            s = "Hello World";
+            IntPtr ptr = Marshal.StringToHGlobalAuto(s);
+            String s2 = Marshal.PtrToStringAuto(ptr);
+            Assert.Equal(s, s2);              
+        }
+        
+        [Fact]
+        public static void BindToMoniker()
+        {
+            String monikerName = null;
+            if(PlatformDetection.IsWindows)
+            {
+                Assert.Throws<ArgumentException>(() => Marshal.BindToMoniker(monikerName));
```

Comment:
I talked with Yi. Checking the null behavior is sufficient since it is not very widely used.